src/Core/Checkout/Cart/Subscriber/CartDeletedSubscriber.php

```diff
@@ -0,0 +1,42 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Cart\Subscriber;
+
+use Shopware\Core\Checkout\Cart\Event\CartDeletedEvent;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\SalesChannel\Context\SalesChannelContextPersister;
+use Shopware\Core\System\SalesChannel\Context\SalesChannelContextService;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+readonly class CartDeletedSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private SalesChannelContextPersister $contextPersister
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            CartDeletedEvent::class => ['onCartDeleted'],
+        ];
+    }
+
+    public function onCartDeleted(CartDeletedEvent $event): void
+    {
+        $salesChannelContext = $event->getSalesChannelContext();
+
+        $this->contextPersister->save(
```

Comment:
We should go a step higher and use the `SalesChannelContextSwitcher` to also dispatch necessary events